lib/rdoc/parser/ruby.rb

```diff
@@ -1470,10 +1470,30 @@ def parse_method_params_and_body container, single, meth, added_container
         end
       end
 
+      coerce_call_seq(meth)
+
       parse_statements container, single, meth
     end
   end
 
+  # Coerce calling sequence if the given method is a binary operator method.
+  def coerce_call_seq(meth)
```

Comment:
We should make this method private and not directly test it, but rather test the encapsulating `parse_method` method, because I could delete the call to `coerce_call_seq` from `parse_method_params_and_body` and all the tests would still pass.

Reply:
I see your point, but there's no testing whatsoever for parse_method_params_and_body.  I don't think it should fall to me to take that on.  I'm happy to test the code I've written, as I've done.

Reply:
I disagree. `coerce_call_seq ` (and arguably `parse_method_params_and_body`) are all private methods (i.e. implementation details) of `RDoc::Parser::Ruby`. Private methods shouldn't be directly tested, therefore it must be tested through `parse_method `.

Reply:
Well, I started out trying to do a small good deed, but it seems that small and good, here, are mutually exclusive.  If there's no objection, I will close this PR and consider it a lesson learned.

Reply:
I apologize if my comment sounds dismissive in any way, that is not my intention, I do think this is a valuable feature. I also think that it is my duty to ensure that the code base remains in a high quality, easy to understand, and well maintained condition, which is the reason for all of these comments. Your code looks fine, but tests are also equally as important since it prevents bugs or regressions from other developers. Therefore it's important that the tests are also of a high quality. A feature, no matter how small, needs to fit in the larger picture. What I'm asking here is to change your test to check that your code fits in the larger picture.

Again, it is not my intention to discourage you from working on this, so I apologize if it seems like such.

Reply:
I'm fine, Peter, with  your comments, impulses, and standards, so no problem there.  Nevertheless, I'm done with this.  There are many other things I can be working on that don't get me into such a tangle.

As always, thanks much for your help, which has been critical in keeping me away from many errors and blunders.

With your permission, I'll close this PR.

Reply:
I don't want to make you feel like it's difficult to contribute code. I can understand that sometimes it feels like there's an endless stream of feedback and criticism, I've certainly felt that before. I want to be clear that I'm writing these comments with best intentions and I'm not trying to be difficult (and I realize I may seem as such).

I'll be sad if you threw out your hard work, especially since you're so close to merging it. I'm willing to provide any help or guidance you need to get this merged. However, if you feel like you certainly don't want to work on this any further, it's up to you to decide if you want to close this PR or not.